wordpress-mobile/WordPress-iOS
# Reader: Close details view when user is blocked

Fixes #22165

When a user is blocked, Reader Details is no longer functional and some actions can cause a crash. More information: https://github.com/wordpress-mobile/WordPress-iOS/issues/22165#issue-2028424959

The view needs to be closed the same way it's done when the site is closed

## To test:

1. Open Reader
2. Discover
3. Select an article 
4. Open the context menu on the header
5. Block user
6. The view should navigate back and the article no longer visible in the list

## Regression Notes
1. Potential unintended areas of impact

It's important that article list updates and it's not possible to interact with blocked user article in any other way

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:


UI Changes testing checklist:
